File: lc-137.go
package main

import (
	"fmt"
)

func singleNumber(nums []int) int {
	var one int = 0
	var two int = 0

	for i := 0; i < len(nums); i++ {
		one = (one ^ nums[i]) & ^two
		two = (two ^ nums[i]) & ^one
	}

	return one + two
}

func main() {
	inpu := []int{1, 1, 1, 2}
	fmt.Println(singleNumber(inpu))
}

File: lc-137.py
#!/opt/homebrew/bin/python3

from typing import List

def singleNumber(nums: List[int]) -> int:
    one, two = 0, 0
    for i in range(0, len(nums)):
        one = (one ^ nums[i]) & ~two
        two = (two ^ nums[i]) & ~one
    return one + two

# inp = [2,2,3,2]
inp = [3,2,3,3,3]

print(str(singleNumber(inp)))
